components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/APIStateChangeWorkflow.java

```diff
@@ -226,6 +232,10 @@ private void updateAPIStatusForWorkflowComplete(String apiId, String status, Str
             String errorMsg = "Couldn't change the status of api ID " + apiId;
             log.error(errorMsg, e);
             throw new APIManagementException(errorMsg, e, ExceptionCodes.APIMGT_LIFECYCLE_EXCEPTION);
+        } catch (GatewayException e) {
+            String message = "Error occurred while changing the state of api ID " + apiId + " in gateway";
```

Comment:
it's better to include the status that he trying to change to log message as well